test/membrane_ffmpeg_swresample_plugin/pts_forward_test.exs

```diff
@@ -0,0 +1,64 @@
+defmodule Membrane.FFmpeg.SWResample.PtsForwardTest do
+  use ExUnit.Case
+
+  import Membrane.ChildrenSpec
+  import Membrane.Testing.Assertions
+
+  alias Membrane.FFmpeg.SWResample.Converter
+  alias Membrane.{RawAudio, Testing}
+
+  test "pts forward test" do
+    input_stream_format = %RawAudio{sample_format: :s16le, sample_rate: 16_000, channels: 2}
+    output_stream_format = %RawAudio{sample_format: :s32le, sample_rate: 32_000, channels: 2}
+
+    # 32 frames * 2048 bytes
+    fixture_path = Path.expand(Path.join(__DIR__, "/../fixtures/input_s16le_stereo_16khz.raw"))
+
+    spec = [
+      child(:source, %Membrane.Testing.Source{output: buffers_from_file(fixture_path)})
+      |> child(:resampler, %Converter{
+        input_stream_format: input_stream_format,
+        output_stream_format: output_stream_format
+      })
+      |> child(:sink, Membrane.Testing.Sink)
+    ]
+
+    pipeline = Testing.Pipeline.start_link_supervised!(spec: spec)
+    assert_start_of_stream(pipeline, :sink)
+    assert_sink_buffer(pipeline, :sink, _buffer)
+
+    Enum.each(0..30, fn index ->
+      assert_sink_buffer(pipeline, :sink, %Membrane.Buffer{pts: out_pts})
+      assert out_pts == index * 31_250_000
+    end)
+
+    assert_sink_buffer(pipeline, :sink, %Membrane.Buffer{pts: 937_500_000})
```

Comment:
why separate assertion for this one?

Reply:
Because of the way converter sets pts. In this exact case, last two buffers have the same pts, so there can be a separate assert outside of the loop, or I would have to make a condition inside the loop above that checks if It's on the last index, to assert with pts the same as previous one. Don't know which solution is prettier 🧐

Reply:
Ok. Why do they have the same PTS?

Reply:
because converter buffers some data while resampling, last two buffers are a bit smaller in size, and both are made from data that came with the same pts. (algorithm that we talked about on slack)